### PR TITLE
💰 Added option to sort by cost

### LINKED ISSUE #215: Sort by cost
Currently our resources are by default sorting by alphabetical order. Add a drop down menu to the left of the search bar and add an option to sort by cost.

https://ant.design/components/select/
https://ant.design/components/dropdown/
image
Can originally just say "sort by: name" (or whatever is auto selected), then can change to "sort by: cost" when that is selected

### PR DESCRIPTION
**Pull request by abetco**
# Pull Request Description

As of now, the resources are being default sorted alphabetically. This PR adds the functionality of being able to sort by cost as well, with an added dropdown selection on the resources page. (Only on desktop, functionality not implemented for mobile).

Cost is sorted as follows, from first to last:
- Free
- $
- $$
- $$$
- No cost information

If relevant, include:
Fixes #215 

image

### CODE REVIEW

File: client/yarn.lock [hunk 1]
```diff
@@ -1898,11 +1898,6 @@ abab@^2.0.0:
   resolved "https://registry.yarnpkg.com/abab/-/abab-2.0.3.tgz#623e2075e02eb2d3f2475e49f99c91846467907a"
```
Comment (abetco) [inline client/yarn.lock:1898]: I have no idea why this stuff is here.

File: client/src/components/desktop/ResourcesFilter.js [hunk 2]
```diff
@@ -31,17 +34,20 @@ function ResourcesFilter(props) {
...
   );
 
   const radio = useCallback(
-    (filterName, filterOptions, value) => {
+    (filterName, filterOptions, value, isSort) => {
       return (
-        <div className="radio-container">
-          <h4 className="title">{filterName}</h4>
+        <div className={isSort ? 'radio-container-sort' : 'radio-container'}>
+          {isSort ? null : <h4 className="title">{filterName}</h4>}
```
Comment (n3a9) [inline client/src/components/desktop/ResourcesFilter.js:50]: What you can do is use [conditional rendering](https://reactjs.org/docs/conditional-rendering.html)

`{!isSort && <h4 className="title">{filterName}</h4>}`

File: client/src/pages/Resources.js [hunk 3]
```diff
@@ -56,6 +58,34 @@ function Resources(props) {
...
   }, []);
 
+  function compareNames(current, next) {
+    const textCurrent = current.name.toUpperCase();
+    const textNext = next.name.toUpperCase();
+    const bool = textCurrent > textNext ? 1 : 0;
+    return textCurrent < textNext ? -1 : bool;
+  }
+
+  function compareCosts(current, next) {
```
Comment (n3a9) [inline client/src/pages/Resources.js:68]: this is fine but one easy might be to just take the array of [free $ $$ $$$] and compare their indices

File: client/yarn.lock (hunk 1 above)
Comment (n3a9) [reply client/yarn.lock:1898]: jsut `yarn upgrade` in the client directory before you push to master, might clear some of these diffs

File: client/src/pages/Resources.js (hunk 3 above)
Comment (abetco) [reply client/src/pages/Resources.js:68]: fixed

**Pull request merged**